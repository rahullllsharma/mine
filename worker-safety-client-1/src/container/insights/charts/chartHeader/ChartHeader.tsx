import type { WorkbookData } from "@/components/fileDownloadDropdown/providers/spreadsheet";

import FilesDownloadDropdown from "@/components/fileDownloadDropdown/FileDownloadDropdown";

type ChartHeaderFileDownloadDropdownProps = {
  chartFilename: string;
  chartData: WorkbookData;
  downloadable?: boolean;
  actionTitle?: string;
};

function HeaderDownloadComponent({
  chartData,
  downloadable = true,
  actionTitle = "",
  chartFilename,
}: ChartHeaderFileDownloadDropdownProps) {
  return (
    <FilesDownloadDropdown
      className="flex-0"
      fileName={chartFilename}
      data={chartData}
      action={{
        title: actionTitle,
        disabled: !downloadable || chartData.length === 0,
      }}
    />
  );
}

type ChartHeaderProps = {
  title: string;
  chartData?: WorkbookData;
  downloadable?: boolean;
  actionTitle?: string;
  chartFilename: string;
};

export default function ChartHeader({
  title,
  ...headerDropdownProps
}: ChartHeaderProps): JSX.Element {
  return (
    <header className="flex flex-nowrap items-center justify-around">
      <h2 className="text-center font-semibold text-xl flex-1">{title}</h2>
      {headerDropdownProps.chartData ? (
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        <HeaderDownloadComponent {...headerDropdownProps} />
      ) : null}
    </header>
  );
}
