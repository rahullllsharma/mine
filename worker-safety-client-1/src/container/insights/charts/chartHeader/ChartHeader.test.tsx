import type { ComponentProps } from "react";
import type { WorkbookData } from "@/components/fileDownloadDropdown/providers/spreadsheet";

import { render, screen } from "@testing-library/react";
import ChartHeader from "./ChartHeader";

const renderComponent = (
  props: ComponentProps<typeof ChartHeader> = {
    title: "title",
    chartData: [],
    chartFilename: "sample-file.ext",
  }
) => {
  return render(<ChartHeader {...props} />, {
    wrapper({ children }) {
      return <>{children}</>;
    },
  });
};

describe(ChartHeader.name, () => {
  const props = {
    title: "header bar title",
    chartFilename: "sample.ext",
    chartData: [["hello", [{ column: "value" }]]] as WorkbookData,
  };

  it("should render correctly", () => {
    const { asFragment } = renderComponent(props);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should match the title passed as the header title", () => {
    renderComponent(props);

    screen.getByRole("heading", {
      name: /header bar title/gi,
      level: 2,
    });
  });

  it("should include the file download dropdown", () => {
    renderComponent(props);

    screen.getByRole("button", {
      name: /download csv\/xlsx files/i,
    });
  });

  describe("when the chart data is empty", () => {
    it("should disable the dropdown action button ", () => {
      const title = "please select a chart";

      renderComponent({
        ...props,
        chartData: [],
        downloadable: true,
        actionTitle: title,
      });

      expect(screen.getByTitle(title)).toBeDisabled();
    });
  });

  describe("when is not downloadable", () => {
    it("should disable the dropdown action button and include a title", () => {
      const title = "please select a chart";
      renderComponent({
        ...props,
        downloadable: false,
        actionTitle: title,
      });

      expect(screen.getByTitle(title)).toBeDisabled();
    });
  });
});
