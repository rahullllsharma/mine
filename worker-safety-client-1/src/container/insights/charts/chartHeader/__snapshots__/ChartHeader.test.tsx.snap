// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartHeader should render correctly 1`] = `
<DocumentFragment>
  <header
    class="flex flex-nowrap items-center justify-around"
  >
    <h2
      class="text-center font-semibold text-xl flex-1"
    >
      header bar title
    </h2>
    <div
      class="relative inline-block z-30 flex-0"
      data-testid="dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        aria-label="Download CSV/XLSX files"
        class="text-xl text-neutral-shade-75 disabled:text-neutral-shade-38 disabled:cursor-not-allowed w-7 h-7 flex justify-center items-center gap-1 rounded
        hover:bg-neutral-light-16 focus:bg-neutral-shade-7 active:bg-neutral-shade-7"
        id="headlessui-menu-button-1"
        title=""
        type="button"
      >
        <i
          aria-hidden="true"
          class="ci-more_horizontal"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
